Add Risk Score & Generate Sbom

nvima opened the pull request:
Added the task "riskScore" which also runs at the end of the "runDepTrackWorkflow".

The configuration is optional, if left empty the task will be skipped.

It can be used with optional attributes "timeout" and "maxRiskScore".
- The "timeout" is needed because the calculation of the Risk Score after the Vex upload needs some time until the API returns the correct value.
- "maxRiskScore" can be configured to check in pipeline if the Risk Score is not higher than the given value, if it is, the Gradle task throws an error.
Without "maxRiskScore" only the RiskScore gets printed without an error.

Tried to include the task in the integration tests, but the default api key of the docker container does not work with the Analyze API route.

----------------------------------------------------------

Added the task "generateSbom" and applied internally the CyclonDx Gradle Plugin.
Also added dependsOn, so that generateSbom always runs before uploadSbom, generateVex..
Modified integrationTests to use the generated SBOM instead of the Test SBOM File.

Adjusted README with new instructions and infos.

Ingwersaft commented on README.md:
```diff
@@ -86,6 +94,12 @@ dependencyTrackCompanion {
+    riskScore{
+        projectName.set(name)
+        projectVersion.set(version)
+        timeout.set(20)
```
Please use kotlin duration. Without any unit of measurement it's unclear if we're talking about seconds, milliseconds or something else ;)

nvima commented:
@Ingwersaft 
Added the task "generateSbom" and applied internally the CyclonDx Gradle Plugin.
Also added dependsOn, so that generateSbom always runs before uploadSbom, generateVex..
Modified integrationTests to use the generated SBOM instead of the Test SBOM File.

Adjusted README with new instructions and infos.
